standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -3960,9 +3926,57 @@ public List<Partition> listPartitionsPsWithAuth(String catName, String db_name,
     return partitions;
   }
 
+  private Integer getNumPartitionsViaOrmPs(Table table, List<String> partVals) throws MetaException {
+    // size is known since it contains dbName, catName, tblName and partialRegex pattern
+    Map<String, String> params = new HashMap<>(4);
```

Comment:
if this method isn't reused elsewhere, I would prefer put this codes into `getJdoResult`.

Reply:
Done.